src/orion/core/worker/trial.py

```diff
@@ -183,12 +189,7 @@ def to_dict(self):
             trial_dictionary[attrname] = list(map(lambda x: x.to_dict(),
                                                   getattr(self, attrname)))
 
-        # Trial object should not be able to set a value
-        # (whatever value) to its `_id` on its own, but is should just
-        # expect that it could have one
-        trial_dictionary.pop('id')  # Pop invalid name for database id key
-        if self.is_registered:
-            trial_dictionary['_id'] = self.id
+        trial_dictionary['_id'] = self.id
```

Comment:
I am in agreement with these changes since `_id` is to be dictated in reverse now!